Refactor components for improved readability and maintainability

sanderPostma opened the pull request:
- Updated function parameter formatting in PublishInfo, DisplayStepCredentials, EditProofRequest, and SortableStep components for consistency.
- Adjusted layout styles in DisplayStepCredentials and LocalFileUpload components for better UX.
- Enhanced CreateScenariosScreen to dynamically display the active persona's scenario.
- Removed unused Sidebar component and related code.
- Updated text input components to improve code clarity.
- Fixed comments in use-onboarding-adapter and use-showcase-adapter hooks for better understanding.
- Deleted unused UI store and utility test files.
- Updated Next.js configuration for image handling.
- Removed unnecessary Helm chart files and dependencies.
- Updated Docker environment variables for clarity and correctness.
- Improved OpenAPI documentation for better schema representation.

nklomp committed:
docs: Readme updates

NimeshNB committed:
Refactor components for improved readability and maintainability

- Updated function parameter formatting in PublishInfo, DisplayStepCredentials, EditProofRequest, and SortableStep components for consistency.
- Adjusted layout styles in DisplayStepCredentials and LocalFileUpload components for better UX.
- Enhanced CreateScenariosScreen to dynamically display the active persona's scenario.
- Removed unused Sidebar component and related code.
- Updated text input components to improve code clarity.
- Fixed comments in use-onboarding-adapter and use-showcase-adapter hooks for better understanding.
- Deleted unused UI store and utility test files.
- Updated Next.js configuration for image handling.
- Removed unnecessary Helm chart files and dependencies.
- Updated Docker environment variables for clarity and correctness.
- Improved OpenAPI documentation for better schema representation.

sanderPostma committed:
reverted spacing-only modified files

sanderPostma committed:
reverted charts stuff

sanderPostma committed:
moved docs

esune commented on resources/architecture.md:
```diff
@@ -0,0 +1,87 @@
+
+
+UI Architecture
+There were errors rendering macro:
+•	An unknown error occurred.
```
Issues with documentation renderer? See below as well.

NimeshNB replied:
The documentation has been updated accordingly.

esune commented on README.md:
`````diff
@@ -1,39 +1,121 @@
+```
+## 📦 Package Management
+
+This project uses **pnpm** and **turbo** for monorepo and package management.
+
+### Make sure pnpm is installed
+
+````shell
+npm -g install pnpm
+````
+
+### Install Dependencies
+
+```bash
+pnpm install
+```
+
+### Build the Project
+
+```bash
+pnpm build
+```
+
+### Run Tests
+
+```bash
+pnpm test
+```
+
+For CI/CD pipelines:
+
+```bash
+pnpm test:ci
+```
+
+### Start the Project
+
+```bash
+pnpm start
+```
`````
This still does not work for me when checking-out a clean repository) no previously created `node_modules`):
- `pnpm test:ci` test fail (most of them)
- `pnpm start` fails with the following error: ` ERR_PNPM_NO_SCRIPT_OR_SERVER  Missing script start or file server.js`

NimeshNB replied:
The README file has been updated with all the required setup steps.

NimeshNB committed:
docs: Update README and environment files for improved setup instructions and configurations

NimeshNB committed:
chore: Update .gitignore to include .env.example and add .env.example file

esune commented on apps/bc-wallet-demo-server/.env.example:
```diff
@@ -1,10 +1,21 @@
+WALLET_ID=000000000000000000000000 # for local development
+TRACTION_DEFAULT_TENANT_ID=000000000000000000000000
+TRACTION_DEFAULT_WALLET_ID=000000000000000000000000
+TRACTION_DEFAULT_API_KEY=000000000000000000000000
```
What's the difference between these two sets of variables?

esune replied:
Still unclear on what these variables are/how they are different from the other similarly-named one.

NimeshNB replied:
Updated values in .env.example

sanderPostma replied:
WALLET_ID is not used anywhere

API_KEY & WALLET_ID are (only) used by the bc-wallet-demo-server backend which are still there from the original server project we did not change much. 

TRACTION_DEFAULT_TENANT_ID, TRACTION_DEFAULT_WALLET_ID, TRACTION_DEFAULT_API_KEY are fallback fields for the Traction adapter. These fields now sit encrypted in Tenant table in the API server and are also supplied (encrypted) to the traction adapter from there. But for development and integration tests they are still useful.

(demo-server should also use the values from the Tenant table in the future, but atm it would need to authenticate as root tenant in Keycloak using client id & secret to be able to get to those. Or maybe even better, integrate all backends into one.)

esune commented on apps/bc-wallet-demo-server/.env.example:
```diff
@@ -1,10 +1,21 @@
+TRACTION_DEFAULT_WALLET_ID=000000000000000000000000
+TRACTION_DEFAULT_API_KEY=000000000000000000000000
 TRACTION_API_URL=https://traction-api.com
 TRACTION_DID=00000000000000000
```
What `did` does this need to be set to? Why is it required?

esune commented on apps/bc-wallet-traction-adapter/.env.example:
```diff
@@ -8,9 +8,13 @@ AMQ_TRANSPORT=tls
+TRACTION_DEFAULT_SHOWCASE_API_URL=http://localhost:5005
+TRACTION_TENANT_SESSION_CACHE_SIZE=1024
+TRACTION_TENANT_SESSION_TTL_MINS=1440
+TRACTION_DEFAULT_TENANT_ID=
+TRACTION_DEFAULT_WALLET_ID=
+TRACTION_DEFAULT_API_KEY=
```
Are these supposed to be the same as in teh other services?

NimeshNB replied:
Yes, they should be consistent across all services.

esune reviewed (commented):
Builds complete correctly and the main issues have been resolved, however I am still having issues when trying to start the services locally both in docker (auth with Keycloak is failing) and natively, where I get the following error after I run `pnpm dev`:
```
 × Failed to add workspace "bc-wallet-showcase-creator" from "apps/bc-wallet-showcase-creator/.next/standalone/apps/bc-wallet-showcase-
  │ creator/package.json", it already exists at "apps/bc-wallet-showcase-creator/package.json"

 ELIFECYCLE  Command failed with exit code 1.
 ```

NimeshNB committed:
docs: Update README with clean build artifacts instructions and enhance clean script in package.json

NimeshNB commented:
> Builds complete correctly and the main issues have been resolved, however I am still having issues when trying to start the services locally both in docker (auth with Keycloak is failing) and natively, where I get the following error after I run `pnpm dev`:
> 
> ```
>  × Failed to add workspace "bc-wallet-showcase-creator" from "apps/bc-wallet-showcase-creator/.next/standalone/apps/bc-wallet-showcase-
>   │ creator/package.json", it already exists at "apps/bc-wallet-showcase-creator/package.json"
> 
>  ELIFECYCLE  Command failed with exit code 1.
>  ```
> ```

I've updated the documentation to include a command for cleaning build artifacts after running `pnpm build`

NimeshNB committed:
chore: Clean up .env.example by removing unused variables

NimeshNB committed:
docs: Enhance README with instructions for running bc-wallet-showcase-creator natively and clarify Docker usage

sanderPostma committed:
Added Keycloak instructions

sanderPostma committed:
Added Keycloak instructions

esune commented on resources/authentication/Showcase builder Keycloak configuration.md:
```diff
@@ -0,0 +1,17 @@
+# Root tenant
+
+For tenant management and token introspection we have a separate client which is configured in the API server module.  
+<br/>![](data:image/png;base64,iVBORw0KGgoAAAANSUhEUgAABXEAAAORCAIAAAA753baAAAAAXNSR0IArs4c6QAAAAlwSFlzAAAOxAAADsQBlSsOGwAA/7VJREFUeF7s3Qk8lPkfB/AH0WlTSjU6qDZd2raUTClKlN2/7T6Ikg6b2HSfOnbbtm07VseqVbYcu0llbavooGwjlu3QNQojjFITNUVGzP/3PDPDEGHcfJ5Xr12e+Z3v32N4vvP7/R4lsVhM4YAABCAAAQhAAAIQgAAEIAABCEAAApUUUK5keiSHAAQgAAEIQAACEIAABCAAAQhAAAK0AGIKuA4gAAEIQAACEIAABCAAAQhAAAIQUEQAMQVF1JAHAhCAAAQgAAEIQAACEIAA ...
```
This image is not showing on the branch, not sure what it is supposed to be. If it is not required please remove.

NimeshNB replied:
Enhanced markdown documentation with relevant images in `png` format

esune commented on resources/authentication/Showcase builder Keycloak configuration.md:
```diff
@@ -0,0 +1,17 @@
+
+Showcase builder front-end tenants should be of type public. For write / update / delete operations the API server will use its own client (showcase-root) to introspect the token created by the public client on the Keycloak server to validate it’s integrity, issuer origin and whether the user has logged out his session in the meantime.
+
+![](data:image/png;base64,iVBORw0KGgoAAAANSUhEUgAABEwAAANoCAIAAABV+ehDAAAAAXNSR0IArs4c6QAAAAlwSFlzAAAOxAAADsMB2mqY3AAA/7VJREFUeF7s3Q9czPcfB/BvNWfhtizCJSvMzSazn0hHdKRoW6v5E1KEYaJpm7HZYjYzw1jYYtHozyaZ1ixKlDWXbjV/zr9DOspRc6QbcVb9Pt/v3dX1T3V1dd29vg+PLd/7/H1+Ppfv+z6f7/dMysvLKRwQgAAEIAABCEAAAhCAAAQMRcDUUDqCfkAAAhCAAAQgAAEIQAACEKAFEORgHkAAAhCAAAQgAAEIQAACBiWAIMeghhOdgQAEIAABCEAAAhCAAAQQ5GAOQAACEIAABCAAA ...
```
Same as above: what is this?

NimeshNB replied:
Enhanced markdown documentation with relevant images in `png` format (it's an screenshot from keycloak setup)

esune commented on apps/bc-wallet-demo-server/.env.example:
```diff
@@ -1,10 +1,16 @@
+
+# cors settings
+CORS_DISABLED=false
+CORS_ALLOW_ORIGINS=https://bcshowcase-ui.dev.nborbit.ca,http://localhost:5003
```
Please avoid referencing deployed environments (especially non-BCGov ones) in the examples. Can you update this (and other) value files to please use resources that are local? I believe this url should be the `http://host.docker.internal:5005`, as an example.

Additionally, I am still getting CORS errors even after setting `CORS_ALLOW_ORIGINS=*`.

NimeshNB replied:
The `. env.example` file has been updated.

Could you specify which application is encountering the CORS error?

esune replied:
The `bc-wallet-shocase-creator` is throwing CORS errors

sanderPostma replied:
This line might help if not there already
```
CORS_ALLOW_HEADERS=Content-Type,Authorization,X-Requested-With,Cache-Control,Pragma,Expires
```

esune commented on README.md:
````diff
@@ -1,31 +1,231 @@
-  
+For each of the apps under the apps/ directory, you'll need to create a ```.env``` file from the corresponding ```.env.example``` template. This ensures all environment variables are set up correctly before running the applications.
+
+```bash
+cp apps/bc-wallet-api-server/.env.example apps/bc-wallet-api-server/.env
+cp apps/bc-wallet-demo-server/.env.example apps/bc-wallet-demo-server/.env
+cp apps/bc-wallet-demo-web/.env.example apps/bc-wallet-demo-web/.env
+cp apps/bc-wallet-showcase-creator/.env.example apps/bc-wallet-showcase-creator/.env
+cp apps/bc-wallet-traction-adapter/.env.example apps/bc-wallet-traction-adapter/.env
+```
````
Is this required even when running in Docker, or only to run the application natively?

NimeshNB replied:
Yes, it's required even when running the application in Docker.

NimeshNB committed:
Refactor code structure for improved readability and maintainability

NimeshNB committed:
fix: Correct step descriptions in showcase creation instructions

esune commented:
@NimeshNB @sanderPostma @getmoin I have opened #291 as parallel PR to this one to test CI/CD and deployments for the application. While it needs to be manually kept in-sync by me, it will work around some of the limitations we have been facing with PRs not being deployable when opened from forks.

Please refer to it for CI/CD issues (build and/or deployment) as it should provide you with access to a running environment for validation if successful. I will try to keep an eye open and sync changes as soon as they happen. This is the same workflow that was used to deploy the current dev/validation environment, so the secrets in the pipeline should be correct (unless the workflows are changed) and the values - if they were to need updating - are picked-up from the default `charts/bc-wallet/values.yaml`

sanderPostma committed:
Merge remote-tracking branch 'bcgov/showcase-builder' into integration/showcase-builder-may21

sanderPostma committed:
Merge remote-tracking branch 'origin/integration/showcase-builder-may21' into integration/showcase-builder-may21

sanderPostma committed:
chore: Fetch full tenant record for buildSendOptions()

sanderPostma committed:
chore: Fetch full tenant record for buildSendOptions()

NimeshNB committed:
Update environment configurations and Docker setup for local development

- Updated README file to update hosts file
- Adjusted PORT settings in .env.example files for bc-wallet-api-server and bc-wallet-demo-web.
- Enhanced CORS settings to allow multiple origins in bc-wallet-api-server.
- Updated Dockerfile for bc-wallet-demo-server to include public assets.
- Modified .env.example for bc-wallet-showcase-creator to use local API URLs for development.

esune commented on README.md:
```diff
@@ -84,33 +84,77 @@ cp docker/dev/.env.example docker/dev/.env
-This will start all services, including the frontend (bc-wallet-showcase-creator) and backend containers.
 
-#### 🛠 Run bc-wallet-showcase-creator Natively
+#### ✅ Solution (Local Dev Only)
+
+Use your operating system's `hosts` file to map the Docker service names to `localhost`.
+
+This will allow your browser to resolve the same hostnames as used by services in Docker Compose.
```
I am not really a fan: this is pretty involved and not very developer friendly - one more moving part to be aware of. What about using next.js rewrites for dev mode? https://nextjs.org/docs/app/api-reference/config/next-config-js/rewrites

esune replied:
You could also look into using [host.docker.internal](https://docs.docker.com/desktop/features/networking/#use-cases-and-workarounds), but a URL rewrite might be the easiest way to keep things clean.

getmoin replied:
Nimesh has added the `host.docker.internal` solution

getmoin committed:
Refactor Dockerfiles across all applications to enhance user permissions  for improved security and consistency in runtime environments.

getmoin committed:
Merge branch 'integration/showcase-builder-may21' of https://github.com/4sure-tech/BC-Wallet-Demo into integration/showcase-builder-may21

NimeshNB committed:
Update API URL configuration in .env.example for Docker compatibility
- Updated README

NimeshNB committed:
feat: Enhance onboarding and scenario management with drag-and-drop functionality and improved step handling

sanderPostma committed:
chore: Dummy OIDC_ROOT_ISSUER_URL, OIDC_ROOT_CLIENT_ID and OIDC_ROOT_CLIENT_SECRET env vars for testing

sanderPostma committed:
Merge branch 'integration/showcase-builder-may21' of https://github.com/4sure-tech/BC-Wallet-Demo into integration/showcase-builder-may21

sanderPostma committed:
chore: Debugging & tuning helm deployment

sanderPostma committed:
chore: fallbackEnv for local dev envs

sanderPostma committed:
chore: fallbackEnv for local dev envs

sanderPostma committed:
chore: fallbackEnv for local dev envs

sanderPostma committed:
chore: Traction adapter enhancements

sanderPostma committed:
chore: fix swapped out NEXT_PUBLIC_ env vars

sanderPostma committed:
Merge pull request #176 from 4sure-tech/integration/showcase-builder-june-4

chore: Debugging & tuning helm deployment

sanderPostma committed:
chore: fix test

sanderPostma committed:
Merge remote-tracking branch 'origin/integration/showcase-builder-may21' into integration/showcase-builder-may21

sanderPostma committed:
chore: added rabbitmq binary chart

sanderPostma committed:
chore: Additional fixes for env vars

sanderPostma committed:
chore: use default CORS setting

sanderPostma committed:
chore: Chart env cleanup

sanderPostma committed:
chore: Chart env cleanup

NimeshNB committed:
feat: Enhance scenario creation and management

- Updated CreateScenariosStepsScreen to include sample actions in steps.
- Modified ProofAttribute component to set the value of the attribute.
- Initialized default values in ScenarioEdit for new scenarios.
- Improved SortableStep by removing unnecessary console logs.
- Added functionality to update showcase status and handle showcase creation in ShowcaseList.
- Enhanced onboarding adapter to handle scenario creation and updates more effectively.
- Refactored onboarding creation logic to merge API and local scenarios.
- Improved persona adapter to manage showcase data and handle persona updates.
- Updated presentation adapter to manage scenarios and steps more efficiently.
- Added new hooks for updating presentation scenarios.
- Enhanced step creation logic to allow for custom actions.
- Updated localization files to include new labels for draft showcases.
- Improved middleware to validate tenant IDs and handle errors gracefully.

NimeshNB committed:
feat: Integrate tenant ID and showcase slug in navigation for onboarding flow

NimeshNB committed:
Merge pull request #178 from 4sure-tech/fix/edit-showcase-issues

feat: Enhance scenario creation and management

NimeshNB committed:
refactor: Improve error handling in tenant validation and fetch functions

getmoin committed:
chore: Refactor environment variables for services and remove deprecated envFrom includes

getmoin committed:
Merge pull request #179 from 4sure-tech/integration/showcase-builder-chart-cleanup

chore: refactor env variables for services and remove envFrom includes

NimeshNB committed:
fix: Update project name in README to match the correct application title

bryce-mcmath reviewed (approved):
I don't see any glaring issues with the front end changes, I will let someone else speak to the devops side of things

i5okie commented on charts/bc-wallet/templates/api-server/deployment.yaml:
```diff
@@ -74,7 +81,28 @@ spec:
+            - name: CORS_ALLOW_ORIGINS
+              value: "https://{{ include "bc-wallet.showcaseCreator.host" . }},https://{{ include "bc-wallet.demoWeb.host" . }}"
+            - name: OIDC_ROOT_CLIENT_ID
+               valueFrom:
```
There is an extra white space, breaking this template.
Please fix bad indentation.

getmoin replied:
Fixed it.

getmoin committed:
fix: Correct indentation for api-server deployment.yaml file

i5okie commented on charts/bc-wallet/templates/demo-server/deployment.yaml:
```diff
@@ -47,27 +47,33 @@ spec:
+                secretKeyRef:
+                  name: {{ .Values.demoServer.secretName | default "bc-wallet-secrets" | quote }}
+                  key: TRACTION_WEBHOOK_SECRET
           {{- with .Values.demoServer.livenessProbe }}
```
This template is missing envFrom render line:
it should go after line 76. before code on line 77.
`{{- include "bc-wallet.renderEnvFrom" (list "apiServer" .) | nindent 10 }}`

getmoin replied:
@i5okie I have added the renderEnvFrom function. Can you check now?

getmoin committed:
fix: Add envFrom rendering for services in deployment templates

i5okie commented:
What is the `OIDC_DEFAULT_TENANT` env variable?
This is the list of environment variables set in `bc-wallet-secrets` secret:
Some of these overlap, as we've renamed some of the environment variables, I believe.

```
API_KEY
ENCRYPTION_KEY
NEXT_AUTH_SECRET
OIDC_CLIENT_ID
OIDC_CLIENT_SECRET
OIDC_ISSUER_URL
OIDC_REALM
OIDC_ROOT_CLIENT_ID
OIDC_ROOT_CLIENT_SECRET
OIDC_ROOT_ISSUER_URL
OIDC_SERVER_URL
OIDC_TRUST_HOST
TENANT_ID
TRACTION_API_URL
TRACTION_DEFAULT_API_KEY
TRACTION_DEFAULT_API_URL
TRACTION_DEFAULT_TENANT_ID
TRACTION_DEFAULT_WALLET_ID
TRACTION_WEBHOOK_SECRET
WALLET_ID
```

the showcase creator pod is failing because `OIDC_DEFAULT_TENANT` does not exist in the secret.

NimeshNB commented:
> What is the `OIDC_DEFAULT_TENANT` env variable? This is the list of environment variables set in `bc-wallet-secrets` secret:
> 
> ```
> API_KEY
> ENCRYPTION_KEY
> NEXT_AUTH_SECRET
> OIDC_CLIENT_ID
> OIDC_CLIENT_SECRET
> OIDC_ISSUER_URL
> OIDC_REALM
> OIDC_ROOT_CLIENT_ID
> OIDC_ROOT_CLIENT_SECRET
> OIDC_ROOT_ISSUER_URL
> OIDC_SERVER_URL
> OIDC_TRUST_HOST
> TENANT_ID
> TRACTION_API_URL
> TRACTION_DEFAULT_API_KEY
> TRACTION_DEFAULT_API_URL
> TRACTION_DEFAULT_TENANT_ID
> TRACTION_DEFAULT_WALLET_ID
> TRACTION_WEBHOOK_SECRET
> WALLET_ID
> ```
> 
> the showcase creator pod is failing because `OIDC_DEFAULT_TENANT` does not exist in the secret.

It's an default tenant with value like this `OIDC_DEFAULT_TENANT=showcase-tenantA`

i5okie commented:
> > What is the `OIDC_DEFAULT_TENANT` env variable? This is the list of environment variables set in `bc-wallet-secrets` secret:
> > ```
> > API_KEY
> > ENCRYPTION_KEY
> > NEXT_AUTH_SECRET
> > OIDC_CLIENT_ID
> > OIDC_CLIENT_SECRET
> > OIDC_ISSUER_URL
> > OIDC_REALM
> > OIDC_ROOT_CLIENT_ID
> > OIDC_ROOT_CLIENT_SECRET
> > OIDC_ROOT_ISSUER_URL
> > OIDC_SERVER_URL
> > OIDC_TRUST_HOST
> > TENANT_ID
> > TRACTION_API_URL
> > TRACTION_DEFAULT_API_KEY
> > TRACTION_DEFAULT_API_URL
> > TRACTION_DEFAULT_TENANT_ID
> > TRACTION_DEFAULT_WALLET_ID
> > TRACTION_WEBHOOK_SECRET
> > WALLET_ID
> > ```
> > 
> > 
> >     
> >       
> >     
> > 
> >       
> >     
> > 
> >     
> >   
> > the showcase creator pod is failing because `OIDC_DEFAULT_TENANT` does not exist in the secret.
> 
> It's an default tenant with value like this `OIDC_DEFAULT_TENANT=showcase-tenantA`

`showcase-tenantA` is the "client id", not "tenant".  `OIDC_CLIENT_ID` which is currently set to `showcase-builder-main`

NimeshNB commented:
> > > What is the `OIDC_DEFAULT_TENANT` env variable? This is the list of environment variables set in `bc-wallet-secrets` secret:
> > > ```
> > > API_KEY
> > > ENCRYPTION_KEY
> > > NEXT_AUTH_SECRET
> > > OIDC_CLIENT_ID
> > > OIDC_CLIENT_SECRET
> > > OIDC_ISSUER_URL
> > > OIDC_REALM
> > > OIDC_ROOT_CLIENT_ID
> > > OIDC_ROOT_CLIENT_SECRET
> > > OIDC_ROOT_ISSUER_URL
> > > OIDC_SERVER_URL
> > > OIDC_TRUST_HOST
> > > TENANT_ID
> > > TRACTION_API_URL
> > > TRACTION_DEFAULT_API_KEY
> > > TRACTION_DEFAULT_API_URL
> > > TRACTION_DEFAULT_TENANT_ID
> > > TRACTION_DEFAULT_WALLET_ID
> > > TRACTION_WEBHOOK_SECRET
> > > WALLET_ID
> > > ```
> > > 
> > > 
> > >     
> > >       
> > >     
> > > 
> > >       
> > >     
> > > 
> > >     
> > >   
> > > the showcase creator pod is failing because `OIDC_DEFAULT_TENANT` does not exist in the secret.
> > 
> > 
> > It's an default tenant with value like this `OIDC_DEFAULT_TENANT=showcase-tenantA`
> 
> `showcase-tenantA` is the "client id", not "tenant". `OIDC_CLIENT_ID` which is currently set to `showcase-builder-main`


`showcase-builder-main` should be set as the `OIDC_ROOT_CLIENT_ID`, and `OIDC_DEFAULT_TENANT` should be set to `showcase-tenantA` to load the default tenant when no tenant ID is provided in the URL.

sanderPostma commented:
> > > > What is the `OIDC_DEFAULT_TENANT` env variable? This is the list of environment variables set in `bc-wallet-secrets` secret:
> > > > ```
> > > > API_KEY
> > > > ENCRYPTION_KEY
> > > > NEXT_AUTH_SECRET
> > > > OIDC_CLIENT_ID
> > > > OIDC_CLIENT_SECRET
> > > > OIDC_ISSUER_URL
> > > > OIDC_REALM
> > > > OIDC_ROOT_CLIENT_ID
> > > > OIDC_ROOT_CLIENT_SECRET
> > > > OIDC_ROOT_ISSUER_URL
> > > > OIDC_SERVER_URL
> > > > OIDC_TRUST_HOST
> > > > TENANT_ID
> > > > TRACTION_API_URL
> > > > TRACTION_DEFAULT_API_KEY
> > > > TRACTION_DEFAULT_API_URL
> > > > TRACTION_DEFAULT_TENANT_ID
> > > > TRACTION_DEFAULT_WALLET_ID
> > > > TRACTION_WEBHOOK_SECRET
> > > > WALLET_ID
> > > > ```
> > > > 
> > > > 
> > > >     
> > > >       
> > > >     
> > > > 
> > > >       
> > > >     
> > > > 
> > > >     
> > > >   
> > > > the showcase creator pod is failing because `OIDC_DEFAULT_TENANT` does not exist in the secret.
> > > 
> > > 
> > > It's an default tenant with value like this `OIDC_DEFAULT_TENANT=showcase-tenantA`
> > 
> > 
> > `showcase-tenantA` is the "client id", not "tenant". `OIDC_CLIENT_ID` which is currently set to `showcase-builder-main`
> 
> `showcase-builder-main` should be set as the `OIDC_ROOT_CLIENT_ID`, and `OIDC_DEFAULT_TENANT` should be set to `showcase-tenantA` to load the default tenant when no tenant ID is provided in the URL.

It's meant for when you open https://pr-291-bc-wallet-showcase-creator-dev.apps.silver.devops.gov.bc.ca/  without the /en/<tenant id> path, it should redirect to https://pr-291-bc-wallet-showcase-creator-dev.apps.silver.devops.gov.bc.ca/en/showcase-tenantA, but I see that is not working for this deployment atm.

sanderPostma commented:
Not longer needed since the merge of the identifier-resolution branch. It allows for adding extra serializers in the HTTPOptions.

i5okie commented:
> > > > > What is the `OIDC_DEFAULT_TENANT` env variable? This is the list of environment variables set in `bc-wallet-secrets` secret:
> > > > > ```
> > > > > API_KEY
> > > > > ENCRYPTION_KEY
> > > > > NEXT_AUTH_SECRET
> > > > > OIDC_CLIENT_ID
> > > > > OIDC_CLIENT_SECRET
> > > > > OIDC_ISSUER_URL
> > > > > OIDC_REALM
> > > > > OIDC_ROOT_CLIENT_ID
> > > > > OIDC_ROOT_CLIENT_SECRET
> > > > > OIDC_ROOT_ISSUER_URL
> > > > > OIDC_SERVER_URL
> > > > > OIDC_TRUST_HOST
> > > > > TENANT_ID
> > > > > TRACTION_API_URL
> > > > > TRACTION_DEFAULT_API_KEY
> > > > > TRACTION_DEFAULT_API_URL
> > > > > TRACTION_DEFAULT_TENANT_ID
> > > > > TRACTION_DEFAULT_WALLET_ID
> > > > > TRACTION_WEBHOOK_SECRET
> > > > > WALLET_ID
> > > > > ```
> > > > > 
> > > > > 
> > > > >     
> > > > >       
> > > > >     
> > > > > 
> > > > >       
> > > > >     
> > > > > 
> > > > >     
> > > > >   
> > > > > the showcase creator pod is failing because `OIDC_DEFAULT_TENANT` does not exist in the secret.
> > > > 
> > > > 
> > > > It's an default tenant with value like this `OIDC_DEFAULT_TENANT=showcase-tenantA`
> > > 
> > > 
> > > `showcase-tenantA` is the "client id", not "tenant". `OIDC_CLIENT_ID` which is currently set to `showcase-builder-main`
> > 
> > 
> > `showcase-builder-main` should be set as the `OIDC_ROOT_CLIENT_ID`, and `OIDC_DEFAULT_TENANT` should be set to `showcase-tenantA` to load the default tenant when no tenant ID is provided in the URL.
> 
> It's meant for when you open https://pr-291-bc-wallet-showcase-creator-dev.apps.silver.devops.gov.bc.ca/ without the /en/ path, it should redirect to https://pr-291-bc-wallet-showcase-creator-dev.apps.silver.devops.gov.bc.ca/en/showcase-tenantA, but I see that is not working for this deployment atm.

Makes sense.
I've added `OIDC_DEFAULT_TENANT` to the secret. Services redeployed without issue.

sanderPostma committed:
chore: fixed tenant fetch

NimeshNB committed:
fix: remove tenant validation check from middleware

NimeshNB committed:
fix: handle invalid tenant by rewriting to /invalid-tenant

sanderPostma committed:
chore: tenant owns issuer & relying party now

sanderPostma committed:
chore: tenant related tests passing again. Updated traction adapter to include tenantId on credential definitions

NimeshNB committed:
feat: integrate tenant fetching and management across onboarding and showcase components

sanderPostma committed:
chore: updated other tests

sanderPostma committed:
chore: updated other tests

sanderPostma committed:
chore: updated other tests

sanderPostma committed:
chore: all tests green

NimeshNB committed:
feat: add persona description to onboarding components and update showcase API

esune commented:
Updated tracking branch, CI/CD completed and updated version is deployed.

NimeshNB committed:
feat: enhance credential import and approval processes with error handling and logging

NB-PrasadKatkar committed:
credential library issues fixed

NimeshNB committed:
feat: Update AppSidebar to display tenant ID and link to tenant-specific homepage

feat: Enhance NewCharacterPage to manage headshot and body image types

fix: Update AccordionGroup to link to the correct test URL

feat: Modify CredentialsForm to handle credential logo types

fix: Adjust CredentialsImport to correctly extract credential definition name

fix: Update DeleteModal to ensure proper z-index for modal display

feat: Extend FileUploadFull to accept SVG files and handle image type updates

feat: Integrate DeleteModal into LandingPage for showcase deletion confirmation

fix: Refine onboarding creation logic to handle persona scenarios correctly

fix: Update StepEditorForm to improve service step validation logic

feat: Enhance publish screen components to support SVG file uploads

fix: Update showcase list to integrate deletion functionality with modal confirmation

fix: Adjust onboarding adapter to handle scenarios without personas

fix: Update persona adapter to manage image types during persona creation

fix: Refactor showcase hook to allow conditional fetching of showcase data

fix: Correct schema ID parsing logic in utility functions

fix: Update common localization strings for clarity

fix: Adjust French localization strings for consistency

feat: Enhance credential import schema validation with regex patterns

fix: Update service manager to improve token handling and debugging

fix: Adjust showcase API service to correctly reference schema version

fix: Improve traction service token handling and debugging output

NB-PrasadKatkar committed:
updated query key

NimeshNB committed:
feat: Enhance onboarding and showcase components with session management and scenario handling

NB-PrasadKatkar committed:
Merge branch 'fix/bug-fixes' into hotfixes/backend

NB-PrasadKatkar committed:
Merge pull request #180 from 4sure-tech/hotfixes/backend

Hotfixes/backend

NimeshNB committed:
fix: Add missing reference for bc-wallet-traction-adapter in tsconfig

NimeshNB committed:
Merge pull request #181 from 4sure-tech/fix/bug-fixes

Fix/bug fixes

esune commented:
@NimeshNB tracking PR was updated, however without the helm environment changes and fixes to get tests to pass things will not be deployed.

NimeshNB committed:
fix: Update test cases to use 'example_name' for credential definitions

NB-PrasadKatkar committed:
restrict schema with same name and version

NimeshNB committed:
feat: Add environment variables for relying party name and wallet QR code URL

NB-PrasadKatkar committed:
predicates implemented in showcase creator

NB-PrasadKatkar committed:
Merge branch 'integration/showcase-builder-may21' of https://github.com/4sure-tech/BC-Wallet-Demo into hotfixes/backend

NB-PrasadKatkar committed:
added multiple predicates to single proof request

NimeshNB committed:
Merge pull request #182 from 4sure-tech/hotfixes/backend

Hotfixes/backend

NimeshNB committed:
Merge pull request #183 from 4sure-tech/fix/bug-fixes

feat: implemented ZKP

NB-PrasadKatkar committed:
fixed test cases

NimeshNB committed:
Merge pull request #184 from 4sure-tech/hotfixes/backend

fixed test cases

NimeshNB committed:
Merge pull request #185 from 4sure-tech/fix/bug-fixes

Fix: test case

NimeshNB committed:
fixed type issue

NB-PrasadKatkar committed:
store  import credential requests in table

NB-PrasadKatkar committed:
job processing on user login

getmoin committed:
Modify resource limits and requests in values.yaml

Updated resource limits and requests for memory and CPU.

NB-PrasadKatkar committed:
fixed API call response 404

NB-PrasadKatkar committed:
removed unused imports and variables

NimeshNB committed:
Add image upload error handling and validation across components

NimeshNB committed:
Merge pull request #186 from 4sure-tech/hotfixes/backend

Hotfixes/backend

NimeshNB committed:
Refactor presentation creation logic and state management

- Moved presentation creation state management to a new store (use-presentation-store.ts).
- Updated usePresentationCreation and usePresentationAdapter hooks to utilize the new store.
- Improved initialization logic to handle showcase slug changes and stale personas.
- Simplified scenario and step management functions, ensuring better state handling and clarity.
- Removed unused imports and code related to the previous state management approach.

NimeshNB committed:
feat: Enhance proof request handling and improve UI components

- Added a utility function to check for requested properties in proof requests.
- Updated createProofRequest, createDeepProofRequest, and createOOBProofRequest to handle empty attributes gracefully.
- Implemented broken image handling in UseCaseItem and StarterCredentials components.
- Integrated wallet fetching in OnboardingPage after completing onboarding and fetching showcase.
- Enhanced proof deletion logic in Section component based on proof completion status.
- Refactored PresentCredentialAction to use an object for proofs and predicates.
- Improved showcase creation and editing components with better validation messages and mandatory fields.
- Updated Instructions component to provide clearer guidance on creating showcases.
- Removed unnecessary showcase status from ShowcaseList.
- Adjusted schema validations for showcase name and completion message length.

NimeshNB committed:
Merge pull request #187 from 4sure-tech/fix/UAT-issues

feat: Enhance proof request handling and improve UI components

NimeshNB committed:
feat: Enhance onboarding and showcase creation experience

- Updated NoSelection component to include additional subtext options for better user guidance.
- Modified FileUploadFull component to handle image deletion and improved text styling.
- Adjusted LocalFileUpload component for consistent text styling.
- Refactored onboarding creation component to improve drag-and-drop functionality and user feedback.
- Enhanced onboarding steps component with clearer instructions and improved text for better user understanding.
- Updated StepCredentialManager to provide instructional text for credential selection.
- Improved StepEditorForm to include instructional text based on the current step.
- Enhanced ShowcaseCreate and ShowcaseEdit components to manage form dirty state and provide user warnings for unsaved changes.
- Introduced useFormDirtyStore hook to manage form dirty state across components.
- Updated onboarding adapter hooks to provide clearer success messages.
- Changed persona adapter hooks to reflect character terminology.
- Updated persona store to initialize with a creating-new state.
- Enhanced localization files for better clarity and user guidance in both English and French.
- Updated showcase schema to enforce character limits on description fields.

NimeshNB committed:
feat: Enhance onboarding and showcase creation experience

- Updated NoSelection component to include additional subtext options for better user guidance.
- Modified FileUploadFull component to handle image deletion and improved text styling.
- Adjusted LocalFileUpload component for consistent text styling.
- Refactored onboarding creation component to improve drag-and-drop functionality and user feedback.
- Enhanced onboarding steps component with clearer instructions and improved text for better user understanding.
- Updated StepCredentialManager to provide instructional text for credential selection.
- Improved StepEditorForm to include instructional text based on the current step.
- Enhanced ShowcaseCreate and ShowcaseEdit components to manage form dirty state and provide user warnings for unsaved changes.
- Introduced useFormDirtyStore hook to manage form dirty state across components.
- Updated onboarding adapter hooks to provide clearer success messages.
- Changed persona adapter hooks to reflect character terminology.
- Updated persona store to initialize with a creating-new state.
- Enhanced localization files for better clarity and user guidance in both English and French.
- Updated showcase schema to enforce character limits on description fields.

NimeshNB committed:
Merge pull request #188 from 4sure-tech/fix/UAT-issues

feat: Enhance onboarding and showcase creation experience

NimeshNB committed:
fix: Handle image upload error and set credential logo type correctly

NimeshNB committed:
fix: Handle image upload error and set credential logo type correctly

NimeshNB committed:
fix: Handle image upload error and set credential logo type correctly

NimeshNB committed:
feat: Updated Helpers text and updated flows

- Added log for webhook (WebhookController.ts)

- Added log for proof request data (PresentationCredentialAction.tsx)

NimeshNB committed:
feat: Updated Helpers text and updated flows

- Added log for webhook (WebhookController.ts)

- Added log for proof request data (PresentationCredentialAction.tsx)

NimeshNB committed:
feat: Add max length validation for character description

NimeshNB committed:
feat: Add max length validation for character description

NimeshNB committed:
Merge pull request #189 from 4sure-tech/fix/UAT-issues

Fix/uat issues

NimeshNB committed:
feat: Add validation for proof request attributes and predicates

- Implemented validation to ensure at least one attribute or predicate is added for each credential in the EditProofRequest component.
- Added checks for predicate values to ensure they are exactly 8 digits (YYYYMMDD) and not '0'.
- Enhanced the ProofAttribute component to display a date format hint and restrict input length for condition values.
- Updated CreateScenariosScreen to conditionally render the delete scenario button based on the number of scenarios.
- Added error handling and success messages in the onboarding adapter for better user feedback.
- Refactored persona and presentation scenario management in the respective hooks to improve state handling and clarity.
- Updated localization files to include new validation error messages for better user experience.

NimeshNB committed:
feat: Add validation for proof request attributes and predicates

- Implemented validation to ensure at least one attribute or predicate is added for each credential in the EditProofRequest component.
- Added checks for predicate values to ensure they are exactly 8 digits (YYYYMMDD) and not '0'.
- Enhanced the ProofAttribute component to display a date format hint and restrict input length for condition values.
- Updated CreateScenariosScreen to conditionally render the delete scenario button based on the number of scenarios.
- Added error handling and success messages in the onboarding adapter for better user feedback.
- Refactored persona and presentation scenario management in the respective hooks to improve state handling and clarity.
- Updated localization files to include new validation error messages for better user experience.

NimeshNB committed:
Merge pull request #190 from 4sure-tech/fix/UAT-issues

feat: Add validation for proof request attributes and predicates

NimeshNB committed:
fix: Standardize button text and loader messages, enhance onboarding scenario handling

NimeshNB committed:
fix: Standardize button text and loader messages, enhance onboarding scenario handling

NimeshNB committed:
Merge pull request #191 from 4sure-tech/fix/UAT-issues

fix: Standardize button text and loader messages, enhance onboarding

NimeshNB committed:
fix: Update success messages for onboarding scenarios

NimeshNB committed:
fix: Update success messages for onboarding scenarios

NB-PrasadKatkar committed:
logger added

NB-PrasadKatkar committed:
added logger

NimeshNB committed:
fix: Simplify proof deletion conditions and enhance logging for received proof

NB-PrasadKatkar committed:
jobs added for create credential, filter failed or pending cred from issuance and proof request list in showcase

NB-PrasadKatkar committed:
logs added

NB-PrasadKatkar committed:
typescript errors fixed

NB-PrasadKatkar committed:
fixed test cases

NB-PrasadKatkar committed:
Merge branch 'integration/showcase-builder-may21' of https://github.com/4sure-tech/BC-Wallet-Demo into hotfixes/backend

NimeshNB committed:
Merge pull request #192 from 4sure-tech/hotfixes/backend

Hotfixes/backend

NimeshNB committed:
fix: update job status filtering to use 'any' type for job payload data

NB-PrasadKatkar committed:
fixed import issues

NB-PrasadKatkar committed:
Merge pull request #193 from 4sure-tech/hotfixes/backend

fixed import issues
